ui-library-add-call-recording/api/onRecordingFileStatusUpdated.cs

```diff
@@ -0,0 +1,111 @@
+            await TransferRecording(e, log);
+            return new OkResult();
+        }
+
+        static async Task TransferRecording(EventGridEvent e, ILogger log)
+        {
+            var serverCallId = ExtractServerCallIDOrDie(e.Subject);
+            var payload = JsonConvert.DeserializeObject<RecordingFileStatusUpdatedPayload>(e.Data.ToString(), new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
+
+            var callingServerClient = new CallingServerClient(Settings.GetACSConnectionString());
+            var storageClient = new BlobStorage(log, Settings.GetRecordingStoreConnectionString(), Settings.GetRecordingStoreContainerName());
+
+            var chunks = payload.RecordingStorageInfo.RecordingChunks;
+            var blobStorePaths = new List<string>();
+            foreach (var chunk in chunks)
+            {
+                var name = blobName(serverCallId, chunk);
+                var inStream = callingServerClient.DownloadStreaming(new Uri(chunk.ContentLocation)).Value;
+                if (await storageClient.UploadFileAsync(name, inStream))
+                {
+                    blobStorePaths.Add(name);
+                    log.LogInformation($"Transferred {name}");
+                }
+            }
+            log.LogInformation($"Uploaded {blobStorePaths.Count} chunks for {serverCallId}");
+        }
+
+        static ActionResult ValidationMessageResponse(EventGridEvent e)
+        {
+            var data = e.Data as JObject;
+            var eventData = data.ToObject<SubscriptionValidationEventData>();
+            var responseData = new SubscriptionValidationResponse
+            {
+                ValidationResponse = eventData.ValidationCode
+            };
+            if (responseData.ValidationResponse != null)
+            {
+                return new OkObjectResult(responseData);
+            }
+            return new OkResult();
+        }
+
+        static bool IsValidationMessage(EventGridEvent e)
+        {
+            return string.Equals(e.EventType, EventTypes.EventGridSubscriptionValidationEvent, StringComparison.OrdinalIgnoreCase);
+        }
+
+        static bool IsRecordingFileStatusUpdatedMessage(EventGridEvent e)
+        {
+            return string.Equals(e.EventType, "Microsoft.Communication.RecordingFileStatusUpdated", StringComparison.OrdinalIgnoreCase);
+        }
+
+        static string ExtractServerCallIDOrDie(string subject)
+        {
+            var match = serverCallIdExtractionPattern.Match(subject);
+            var serverCallId = match.Groups["serverCallId"].Value;
+            if (string.IsNullOrEmpty(serverCallId))
+            {
+                throw new Exception($"Failed parse serverCallId from {subject}");
+            }
+            return serverCallId;
+        }
+
+        static string blobName(string serverCallId, RecordingChunk chunk)
+        {
+            return $"call_{serverCallId}/chunk_{chunk.DocumentId}_{chunk.Index}.mov";

```

Comment:
nit: I love the utility of this name, but have you considered something jazzier 🕺
```suggestion
            return $"Call Recording 😍 yyyy-mm-dd HH:MM:SS ({chunk.Index})";
```

(only slightly /s 😄)

Reply:
I'm definitely not naming a file in blobstore with the loving-eyes emoji.

But... this is a great suggestion for what's listed in the app frontend.